pkg/influx/errors.go

```diff
@@ -26,42 +21,50 @@ func tryUnwrap(err error) error {
 	return err
 }
 
+// handleError tries to extract an errorx.Error from the given error, logging
+// and setting the http response code as needed. All non-errorx errors are
+// considered internal errors. Please do not try to fix error categorization in
+// this function. All client errors should be categorized as an errorx at the
+// site where they are thrown.
 func handleError(w http.ResponseWriter, r *http.Request, logger log.Logger, err error) {
 	var statusCode int
+	var httpErrString string
 	var errx errorx.Error
 	switch {
-	case errors.Is(err, context.Canceled):
-		statusCode = StatusClientClosedRequest
-		level.Error(logger).Log("msg", "request cancelled", "err", err)
 	case errors.As(err, &errx):
-		switch statusCode = errx.HTTPStatusCode(); statusCode {
-		case http.StatusBadRequest:
-			level.Warn(logger).Log("msg", errx.Message(), "response_code", statusCode, "err", tryUnwrap(errx))
-		default:
-			level.Error(logger).Log("msg", errx.Message(), "response_code", statusCode, "err", tryUnwrap(errx))
-		}
+		httpErrString = errx.Message()
+		statusCode = errx.HTTPStatusCode()
+		err = errx
 	case errors.Is(err, context.DeadlineExceeded) || isGRPCTimeout(err):
+		httpErrString = "network timeout"
 		statusCode = http.StatusGatewayTimeout
-		level.Error(logger).Log("msg", "response timeout", "err", err)
+	case errors.Is(err, context.Canceled):
+		// Note: It seems unlikely this can happen other than as a timeout, so we
+		// should call it an internal error.
+		httpErrString = "request cancelled"
+		statusCode = http.StatusInternalServerError
 	case isNetworkTimeout(err):
```

Comment:
The isGrpcTimeout method at the bottom of the file can be deleted

Reply:
done

Reply:
I restored references to this function so we do need it again